components/Account/Memberships/Claim.js

```diff
@@ -243,36 +253,18 @@ class ClaimMembership extends Component {
       'memberships/claim/addendum'
     ], undefined, '')
 
-    if (polling) {
-      return (
-        <Poller
-          tokenType={signInResponse.tokenType}
-          phrase={signInResponse.phrase}
-          email={values.email}
-          alternativeFirstFactors={signInResponse.alternativeFirstFactors}
-          onCancel={() => {
-            this.setState(() => ({
-              polling: false,
-              loading: false
-            }))
-          }}
-          onTokenTypeChange={(altTokenType) => {
-            this.setState(() => ({
-              polling: false
-            }))
-            this.claim(altTokenType)
-          }}
-          onSuccess={(me) => {
-            this.setState(() => ({
-              polling: false
-            }))
-            this.claim()
-          }} />
-      )
-    }
-    if (loading) {
-      return <Loader loading message={t('memberships/claim/loading')} />
-    }
+    const {
+      tokenType = false,
+      phrase = false,
+      alternativeFirstFactors = []
+    } = signInResponse
+
+    const alternativeFirstFactorsIncludingEmailCode =
+      Array.from(new Set(
+        alternativeFirstFactors
+          .concat(['EMAIL_CODE'])
```

Comment:
Stupid question:
Could the backend return 'EMAIL_CODE' so you don't have to mix it in here? (I might be missing something though)

Reply:
Not a stupid question.

This part is indeed an (intentional) hack to prevent regular login components not to provide `EMAIL_CODE` just yet.

If backend provides it as an alternative, regular login components would render it as an alternative.